tests/background/uri-info-test.js

```diff
@@ -1,46 +1,17 @@
+import { BadgeUriError } from '../../src/background/errors';
 import * as uriInfo from '../../src/background/uri-info';
-//import { toResult } from '../promise-util';
 import settings from '../settings.json';
 
 describe('background/uri-info', () => {
-  const badgeURL = `${settings.apiUrl}/badge`;
-  let fetchStub;
-
-  beforeEach(() => {
-    fetchStub = sinon.stub(window, 'fetch').resolves(
-      new Response('{"total": 1}', {
-        status: 200,
-        headers: {},
-      })
-    );
-  });
-
-  afterEach(() => {
-    window.fetch.restore();
-  });
-
-  describe('getAnnotationCount', () => {
-    describe('fetching data from badge endpoint', () => {
-      it('sends the correct fetch request', () => {
-        return uriInfo.getAnnotationCount('http://tabUrl.com').then(() => {
-          assert.equal(fetchStub.callCount, 1);
-          assert.deepEqual(fetchStub.lastCall.args, [
-            badgeURL + '?uri=http%3A%2F%2FtabUrl.com',
-            { credentials: 'include' },
-          ]);
-        });
-      });
-
-      it('urlencodes the URL appropriately', () => {
-        return uriInfo
-          .getAnnotationCount('http://foo.com?bar=baz qüx')
-          .then(() => {
-            assert.equal(fetchStub.callCount, 1);
-            assert.equal(
-              fetchStub.lastCall.args[0],
-              badgeURL + '?uri=http%3A%2F%2Ffoo.com%3Fbar%3Dbaz+q%C3%BCx'
-            );
-          });
+  describe('uriForBadgeRequest', () => {
+    ['http://.com', 'https://', 'dummy.com'].forEach(badURI => {
+      it('throws for invalid URL', () => {
+        try {
+          uriInfo.uriForBadgeRequest(badURI);
+        } catch (error) {
+          assert.instanceOf(error, TypeError);
+          assert.equal(error.message, "Failed to construct 'URL': Invalid URL");
```

Comment:
The text of this message is not guaranteed across browsers or browser versions, so we may need to update this in future.

Reply:
You are right, I will just check for the type of error that is consistent across browsers.